src/routes/data-decode/__tests__/data-decode.e2e-spec.ts

```diff
@@ -30,62 +29,150 @@ describe('Data decode e2e tests', () => {
+              to: '0x1ea33eb00F2C2F00e1021FD7E9dD22154C82b06F',
+              value: '1000000000000000000',
+              data: null,
+              dataDecoded: null,
+            },
+            {
+              operation: 0,
+              to: '0x844d2C79C4a721CBe153B092bA75B4c1E7Cb2BC3',
+              value: '1000000000000000000',
+              data: null,
+              dataDecoded: null,
+            },
+            {
+              operation: 0,
+              to: '0x182FF57E69eC50eFF8fc4a9E19e4d02D75061D32',
+              value: '1000000000000000000',
+              data: null,
+              dataDecoded: null,
+            },
+            {
+              operation: 0,
+              to: '0xB07d074825596798f6e127f86825AAFaA81Cdd7e',
+              value: '1000000000000000000',
+              data: null,
+              dataDecoded: null,
+            },
+            {
+              operation: 0,
+              to: '0xd39d0a6b980218038C4675310399cecEbb54DE60',
+              value: '1000000000000000000',
+              data: null,
+              dataDecoded: null,
+            },
+          ],
         },
-      ),
-    );
-    const expectedResponse: DataDecoded = JSON.parse(
-      readFileSync(
-        'src/routes/data-decode/__tests__/resources/data-decode-expected-response.json',
-        {
-          encoding: 'utf-8',
-        },
-      ),
-    );
+      ],
+    };
 
     await request(app.getHttpServer())
       .post(`/v1/chains/${chainId}/data-decoder`)
-      .send(requestBody)
+      .send(getDataDecodedDto)
       .expect(200)
       .then(({ body }) => {
         expect(body).toEqual(expectedResponse);
       });
   });
 
   it('POST /data-decoder should throw a validation error', async () => {
-    const requestBody: GetDataDecodedDto = JSON.parse(
-      readFileSync(
-        'src/routes/data-decode/__tests__/resources/data-decode-request-body.json',
-        {
-          encoding: 'utf-8',
-        },
-      ),
-    );
+    const getDataDecodedDto = getDataDecodedDtoBuilder().build();
 
     await request(app.getHttpServer())
       .post(`/v1/chains/${chainId}/data-decoder`)
-      .send({ ...requestBody, to: faker.number.int() })
+      .send({ ...getDataDecodedDto, to: faker.number.int() })
       .expect(400)
       .expect({ message: 'Validation failed', code: 42, arguments: [] });
   });
 
   it('POST /data-decoder should throw a validation error (2)', async () => {
-    const requestBody: GetDataDecodedDto = JSON.parse(
-      readFileSync(
-        'src/routes/data-decode/__tests__/resources/data-decode-request-body.json',
-        {
-          encoding: 'utf-8',
-        },
-      ),
-    );
+    const getDataDecodedDto = getDataDecodedDtoBuilder().build();
 
     await request(app.getHttpServer())
       .post(`/v1/chains/${chainId}/data-decoder`)
-      .send({ ...requestBody, to: faker.string.alphanumeric() })
+      .send({ ...getDataDecodedDto, to: faker.string.alphanumeric() })
```

Comment:
```suggestion
      .send({ ...getDataDecodedDto, to: faker.finance.ethereumAddress() })
```

Reply:
I think the goal of this test is to test a non-ethereum address (as it should fail).

However the validation that we do doesn't really check for an ethereum address but the following regex: `'0[x][0-9a-fA-F]+'`